nobrainer/models/meshnetvwn.py

```diff
@@ -0,0 +1,284 @@
+        )
+
+    with tf.variable_scope('logits'):
+        logits = vwn_conv.conv3d(
+            inputs=outputs, filters=params['n_classes'], kernel_size=(1, 1, 1),
+            padding='SAME', activation=None, is_mc=is_mc
+        )
+
+    predicted_classes = tf.argmax(logits, axis=-1)
+
+    if mode == tf.estimator.ModeKeys.PREDICT:
+        predictions = {
+            'class_ids': predicted_classes,
+            'probabilities': tf.nn.softmax(logits),
+            'logits': logits,
+        }
+        export_outputs = {
+            'outputs': tf.estimator.export.PredictOutput(predictions)}
+        return tf.estimator.EstimatorSpec(
+            mode=mode,
+            predictions=predictions,
+            export_outputs=export_outputs)
+
+    if params['prior_path'] != None:
+        prior_np = np.load(params['prior_path'])
+    
+    with tf.variable_scope("prior"):
+        i=-1
+        for v in tf.get_collection('ms'):
+            i += 1
+            if params['prior_path'] == None:
+                tf.add_to_collection('ms_prior',tf.Variable(tf.constant(0, dtype = v.dtype, shape = v.shape),trainable = False))
+            else:
+                tf.add_to_collection('ms_prior',tf.Variable(tf.convert_to_tensor(prior_np[0][i], dtype = tf.float32),trainable = False))
+        
+        ms = tf.get_collection('ms')
+        ms_prior = tf.get_collection('ms_prior')
+
+        print(len(ms))
+        i=-1
+        for v in tf.get_collection('ms'):
+            i += 1
+            if params['prior_path'] == None:
+                tf.add_to_collection('sigmas_prior',tf.Variable(tf.constant(1, dtype = v.dtype, shape = v.shape),trainable = False))
+            else:
+                tf.add_to_collection('sigmas_prior',tf.Variable(tf.convert_to_tensor(prior_np[1][i], dtype = tf.float32),trainable = False))
+
+        sigmas = tf.get_collection('sigmas')
+        sigmas_prior = tf.get_collection('sigmas_prior')
+     
+    nll_loss = tf.reduce_mean(tf.nn.sparse_softmax_cross_entropy_with_logits(labels=labels, logits=logits))
+    tf.summary.scalar('nll_loss', nll_loss)
+    
+    l2_loss = tf.add_n([tf.reduce_sum((tf.square(ms[i] - ms_prior[i])) / ((tf.square(sigmas_prior[i]) + 1e-8) * 2.0)) for i in range(len(ms))], name = 'l2_loss')
+    tf.summary.scalar('l2_loss', l2_loss)
+    
+    
+    sigma_squared_loss = tf.add_n([tf.reduce_sum(tf.square(sigmas[i]) / ((tf.square(sigmas_prior[i]) + 1e-8) * 2.0)) for i in range(len(sigmas))],name = 'sigma_squared_loss')
+    tf.summary.scalar('sigma_squared_loss', sigma_squared_loss)
+    
+    log_sigma_loss = tf.add_n([tf.reduce_sum(tf.log(v+1e-8)) for v in sigmas],name='log_sigmas_loss')
+    tf.summary.scalar('log_sigma_loss', log_sigma_loss)
+
+    n_examples = tf.constant(params['n_examples'],dtype=ms[0].dtype)
+    tf.summary.scalar('n_examples', n_examples)
+    
+    if not params['only_kld']:
+        loss = nll_loss + (l2_loss + sigma_squared_loss - log_sigma_loss) / (n_examples*256*256*256)
```

Comment:
should the `(n_examples*256*256*256)` be hardcoded? or should the value depend on the shape of the input (sub)volume?

Reply:
Yes, it should be set to the size of the full volume (not subvolume) shape.